packages/components/src/DataTable/Column/DataTableCell.tsx

```diff
@@ -40,46 +41,64 @@ export interface DataTableCellProps
   size?: DataTableColumnSize
 }
 
-const DataTableCellLayout: FC<DataTableCellProps> = ({
-  children,
-  description,
-  indicator,
-  size,
-  ...props
-}) => {
-  let content =
-    size && size !== 'nowrap' ? <Truncate>{children}</Truncate> : children
-
-  if (description) {
-    content = (
-      <SpaceVertical gap="xxxsmall">
-        <span>{content}</span>
-        {description && (
-          <Paragraph fontSize="xsmall" color="subdued" truncate>
-            <Truncate>{description}</Truncate>
-          </Paragraph>
-        )}
-      </SpaceVertical>
-    )
+const DataTableCellLayout = forwardRef(
+  (props: DataTableCellProps, forwardedRef: Ref<HTMLElement>) => {
+    const { children, description, indicator, size } = props
+
+    let content =
+      size && size !== 'nowrap' ? <Truncate>{children}</Truncate> : children
+
+    const ref = useRef<HTMLTableDataCellElement>(null)
+    const forkedRef = useForkedRef(ref, forwardedRef)
 
-    if (indicator) {
+    useEffect(() => {
+      const element = ref?.current
+      const button = element?.querySelector('button')
+      const link = element?.querySelector('a')
+      const input = element?.querySelector('input')
```

Comment:
Could this be made more efficient by looking for them in a single `querySelector`? e.g.:

```suggestion
      const tabIndexElement = element?.querySelector('button, a, input')
```

What would happen if there were more than one button, link or input? 

```suggestion
      const tabIndexElements = element?.querySelectorAll('button, a, input')
```